src/react-native-animated-charts/src/charts/linear/ChartPath.tsx

```diff
@@ -0,0 +1,381 @@
+
+    const translationX = useSharedValue<number | null>(null);
+    const translationY = useSharedValue<number | null>(null);
+
+    const setOriginData = useWorkletCallback(
+      (path: PathData, index?: number) => {
+        if (!path.data.length) {
+          return;
+        }
+
+        if (typeof index === 'undefined') {
+          originalX.value = '';
+          originalY.value = '';
+          return;
+        }
+
+        originalX.value = path.data[index].x.toString();
+        originalY.value = path.data[index].y.toString();
+      },
+      []
+    );
+
+    const resetGestureState = useWorkletCallback(() => {
+      originalX.value = '';
+      originalY.value = '';
+      isActive.value = false;
+      pathOpacity.value = withTiming(
+        1,
+        timingFeedbackConfig || timingFeedbackDefaultConfig
+      );
+      translationX.value = null;
+      translationY.value = null;
+    }, []);
+
+    useEffect(() => {
+      runOnUI(() => {
+        'worklet';
+        if (currentPath) {
+          setOriginData(currentPath);
+        }
+
+        if (currentPath?.path === previousPath?.path) {
+          return;
+        }
+
+        if (progress.value !== 0 && progress.value !== 1) {
+          cancelAnimation(progress);
+        }
+
+        progress.value = 0;
+
+        // this stores an instance of d3-interpolate-path on worklet side
+        // it means that we don't cross threads with that function
+        // which makes it super fast
+        if (previousPath && currentPath) {
+          const d3Interpolate = requireOnWorklet('d3-interpolate-path');
+
+          interpolatorWorklet().value = d3Interpolate.interpolatePath(
+            previousPath.path,
+            currentPath.path
+          );
+
+          progress.value = withDelay(
+            Platform.OS === 'ios' ? 0 : 100,
+            withTiming(1, timingAnimationConfig || timingAnimationDefaultConfig)
+          );
+        } else {
+          interpolatorWorklet().value = undefined;
+          progress.value = 1;
+        }
+      })();
+      // you don't need to change timingAnimationConfig that often
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [
+      currentPath,
+      previousPath,
+      interpolatorWorklet,
+      progress,
+      setOriginData,
+    ]);
+
+    useAnimatedReaction(
+      () => ({ x: translationX.value, y: translationY.value }),
+      values => {
+        if (
+          !currentPath ||
+          !currentPath.parsed ||
+          progress.value === 0 ||
+          values.x === null ||
+          values.y === null
+        ) {
+          return;
+        }
+
+        const yForX = redash.getYForX(currentPath.parsed, Math.floor(values.x));
+
+        if (yForX !== null) {
+          positionY.value = yForX;
+        }
+
+        positionX.value = values.x;
+
+        const index = least(currentPath.points.length, i => {
+          if (typeof i === 'undefined' || values.x === null) {
```

Comment:
```suggestion
          if (i === undefined || values.x === null) {
```

There's no chance that i is not defined is the scope, so we can compare this "normally" I think

Reply:
I would use `typeof` here anyway since it's just a consistent thing with all of the checks for types.